add AmsetFW which has new RunAmset and AmsetToDb tasks

albalu opened the pull request:
## Summary

* I used AmsetFW by adding it after a "nscf uniform" firework. It uses the vasprun.xml from this calculation similar to BoltztrapFW; it also has RunAmset and AmsetToDb very similar to BoltztrapFW 
* AmsetToDb: the mobility and seebeck values are stored as list of jsons that contain the carrier concentration (c) and the temperature (T) at which the property is calculated as well as the actual vector ("vec") and average ("avg") of the calculated vector.
* unlike BoltzTrapFW, Amset is not under pymatgen and is not and should not be an atomate requirement. Therefore, the way I addressed this is to import Amset right under run_task as opposed to the top of the file (hence those who don't have Amset installed won't get ImportError).

computron commented on atomate/vasp/firetasks/parse_outputs.py:
```diff
@@ -259,6 +259,93 @@ def run_task(self, fw_spec):
+
+    Optional params:
+        db_file (str): path to file containing the database credentials.
+            Supports env_chk. Default: write data to JSON file.
```
filename parameter is not documented or even used!

computron commented on atomate/vasp/firetasks/parse_outputs.py:
```diff
@@ -259,6 +259,93 @@ def run_task(self, fw_spec):
+
+    def run_task(self, fw_spec):
+        from amset.core import Amset
+        amset = Amset.from_file()
```
confusing that this is an empty arg ...

computron commented on atomate/vasp/firetasks/parse_outputs.py:
```diff
@@ -259,6 +259,93 @@ def run_task(self, fw_spec):
+        d["created_at"] = datetime.utcnow()
+        d["tags"] = list(set(d.get("tags", []) + fw_spec.get("tags", [])))
+
+        # transform mobility and Seebeck coefficient to list of dicts:
```
might be better to move this into an internal function so that run_task stays relatively clean and high level, and so the documentation for the function has clearly defined input and output formats

computron commented on atomate/vasp/firetasks/run_calc.py:
```diff
@@ -241,6 +242,60 @@ def run_task(self, fw_spec):
+@explicit_serialize
+class RunAmset(FiretaskBase):
+    """
+    Run Amset directly. Requires vasprun.xml to be in current dir.
```
There should be an option to give path to the vasprun.xml file

computron replied:
(unless this is a BoltzTrap limitation, in which case maybe write in the docs that this is the reason)

computron commented on atomate/vasp/firetasks/run_calc.py:
```diff
@@ -241,6 +242,60 @@ def run_task(self, fw_spec):
+    Run Amset directly. Requires vasprun.xml to be in current dir.
+
+    Required params:
+        material_params (dict): the static dielectric constant (epsilon_s) is
```
Tell people where to find more information about what goes inside this dictionary. Very unclear what else can go in and what happens if more or less info is specified

computron commented on atomate/vasp/firetasks/run_calc.py:
```diff
@@ -241,6 +242,60 @@ def run_task(self, fw_spec):
+        performance_params (dict): parameters related to convergence, speed,
+            etc. See Amset documentation for key options
+        temperatures ([float]): the temperatures (in Kelvin)
+        doping: ([float]) doping levels you want to compute
```
document all optional_params, including kgrid_tp, write_outputs, write_inputs

computron commented on atomate/vasp/firetasks/run_calc.py:
```diff
@@ -241,6 +242,60 @@ def run_task(self, fw_spec):
+    def run_task(self, fw_spec):
+        from amset.core import Amset
+        material_params = self.get("material_params")
+        model_params = self.get("model_params", None)
```
why sometimes None and sometimes ``{}`` (as in the next line)

computron commented on atomate/vasp/firetasks/run_calc.py:
```diff
@@ -241,6 +242,60 @@ def run_task(self, fw_spec):
+        material_params = self.get("material_params")
+        model_params = self.get("model_params", None)
+        perfs = self.get("performance_params", {})
+        perfs["max_nvalleys"] = perfs.get("max_nvalleys", 3)
```
make sure the defaults work well for most materials. Although ideally people should fully read the Amset documentation and set these parameters intelligently, the reality is that 90% of people will just run the AmsetFW without thinking about it and report that as the "amset result". So make sure that these defaults give good numbers as they will be representative of what people report as the Amset value.

computron commented on atomate/vasp/firetasks/run_calc.py:
```diff
@@ -241,6 +242,60 @@ def run_task(self, fw_spec):
+        kgrid_tp = self.get("kgrid_tp", "coarse")
+        write_outputs = self.get("write_outputs", True)
+        write_inputs = self.get("write_inputs", True)
+        timeout = self.get("timeout", 10)
```
is this hours? This is an unconventional unit for timeout (usually minutes or seconds). Might want to call the parameter ``timeout_hours``.

albalu commented:
Thanks @computron , I addressed all of these comments except moving this part in AmsetToDb to Amset itself: "transform mobility and Seebeck coefficient to list of dicts" 
Instead I added a todo. The reasons is that I won't be able to properly test the result of this moving in a short amount of time.

computron commented:
Ok although I meant just having it be an internal function of the same class, not moving it to the amset codebase (unless it's useful there). The former should hopefully just take a few moments, it's just copy pasting the code into its own function in that class and creating a docstring for that function.

computron commented:
(also the unit tests)

albalu commented:
ah sorry I misunderstood the internal function. I fixed that. Regarding unit-tests, it would be great if we could defer that until Amset is pip installable and this FW gets used by more people than just myself so we find at least one edge case.

computron commented:
sorry, no unit test, no merge into master

computron commented:
This PR is quite stale and likely will be rewritten from scratch given all the various AMSET updates.

I am closing this.

@utf FYI in case you find any of this code useful
